### PR TITLE
Update descriptions of email BBEs

### PR DESCRIPTION
**Pull request by ayeshLK**
Part of [#3837](https://github.com/ballerina-platform/ballerina-standard-library/issues/3837)

### CODE REVIEW

File: examples/receive-email-using-client/receive_email_using_client.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Email client - Receive email
 
-The email client is used to receive (with POP3 or IMAP4) emails using the SSL or STARTTLS protocols. This sample includes receiving emails with default configurations over SSL using the default ports via POP3. Since, IMAP4 has similar syntax we could replace `POP3 client` with `IMAP client`.
+An email server supports either POP3 or IMAP as mail access protocol. The Ballerina `email` module provides support for POP3 and IMAP mail access clients to read data from an email server. Configure either an `email:PopClient` or an `email:ImapClient` and invoke `receiveMessage` client function to retrieve data from the server.
```
Comment (shafreenAnfar) [inline examples/receive-email-using-client/receive_email_using_client.md:3]: I believe the example is written using `email:PopClient`. So description should be based on pop client.

The `email:PopClient` can receive emails via POP3 protocol. To connect to the server it requires hostname, username and password as the mandatory arguments. Once connected, `receiveMessage` method is used receive email from the server. It is a blocking method with a default timeout. 

> Note: The Ballerina `email` module also provides `email:ImapClient`, which can be used in a similar fashion. The only difference is the `email:ImapClient` uses the IMAP protocol for the communication.

File: examples/send-email/send_email.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # Email client - Send email
 
-The email connector is used to send (with SMTP) emails using the SSL or STARTTLS protocols. This sample includes sending emails with default configurations over SSL using the default ports.
+The `email:SmtpClient` connects to a given SMTP server, and then sends emails to the server. An `email:SmtpClient` is created by providing host-name and required credentials.  Once connected, `sendMessage` or `send` methods are used to send emails to the server. An `email:Message` record, which contains the required information about the email can be passed to the `sendMessage` method. If additional information is not required, the `send` method can be used only with mandatory parameters such as `to`, `from`, `subject`, and `body`.
```
Comment (chamil321) [inline examples/send-email/send_email.md:3]:
```suggestion
The `email:SmtpClient` connects to a given SMTP server to send emails. An `email:SmtpClient` is created by providing the hostname and required credentials.  Once connected, the `sendMessage` or the `send` methods are used to send emails. An `email:Message` record, which contains the required information about the email can be passed to the `sendMessage` method. If additional information is not required, the `send` method can be used only with mandatory parameters such as `to`, `from`, `subject`, and `body`.
```

File: examples/receive-email-using-client/receive_email_using_client.md (hunk 1 above)
Comment (ayeshLK) [reply examples/receive-email-using-client/receive_email_using_client.md]: Fixed with:
- [8cc402cceb100804c7c5451081ef4f0701dc64e1](https://github.com/ballerina-platform/ballerina-distribution/pull/3863/commits/8cc402cceb100804c7c5451081ef4f0701dc64e1)
- [9f3b4631737750fdf69b10c4dd38f76431414582](https://github.com/ballerina-platform/ballerina-distribution/pull/3863/commits/9f3b4631737750fdf69b10c4dd38f76431414582)

File: examples/receive-email-using-client/receive_email_using_client.bal [hunk 3]
```diff
@@ -1,25 +1,19 @@
...
+    do {
+        while true {
+            // Reads the first unseen email received by the POP3 server. 
+            // `()` is returned when there are no new unseen emails. 
+            email:Message? email = check imapClient->receiveMessage(timeout = 30);
+            if email is email:Message {
+                log:printInfo("Received an email", subject = email.subject, body = email?.body);
+            }
+        }
+    } on fail var e {
```
Comment (shafreenAnfar) [inline examples/receive-email-using-client/receive_email_using_client.bal:15]: e -> err

File: examples/receive-email-using-client/receive_email_using_client.md [hunk 4]
```diff
@@ -1,18 +1,20 @@
 # Email client - Receive email
 
-The email client is used to receive (with POP3 or IMAP4) emails using the SSL or STARTTLS protocols. This sample includes receiving emails with default configurations over SSL using the default ports via POP3. Since, IMAP4 has similar syntax we could replace `POP3 client` with `IMAP client`.
+The `email:ImapClient` can receive emails from an email server via IMAP protocol. An `email:ImapClient` can be created by providing the hostname, username, and password. Once connected, `receiveMessage` method is used to receive emails from the server. It is a blocking method with a timeout. If the email server supports both POP3 and IMAP, it is recommended to use the IMAP as it provides the ability to manage emails using multiple devices or email clients, while allowing access to emails that are already read.
```
Comment (shafreenAnfar) [inline examples/receive-email-using-client/receive_email_using_client.md:3]:
```suggestion
The `email:ImapClient` can receive emails from an email server via IMAP protocol. An `email:ImapClient` can be created by providing the hostname, username, and password. Once connected, `receiveMessage` method is used to receive emails from the server. It is a blocking method with a timeout. When the email server supports both POP3 and IMAP, it is recommended to use the IMAP as it provides the ability to manage emails using multiple devices or email clients, while allowing access to emails that are already read. 
```

File: examples/receive-email-using-service/receive_email_using_service.md [hunk 5]
```diff
@@ -1,6 +1,10 @@
 # Email service - Receive email
 
-The email service is used to receive (with POP3 or IMAP4) emails using the SSL or STARTTLS protocols. This sample includes receiving emails service attached to a POP3 listener with default configurations over SSL using the default ports. To use IMAP4 the user can use `IMAP listener`.
+The `email:Service` received messages from an email server via IMAP4 using the `email:ImapListener`.
+
+An `email:Service` which is attached to an `email:ImapListener` can receive emails from an email server via IMAP protocol. An `email:ImapListener` can be initialized by providing the hostname, username, and password. Once connected, `onMessage` method will be invoked whenever an email is read from the email server. If there is an error while reading the data from the server, `onError` method will be invoked with relevant error details. If the email server supports both POP3 and IMAP, it is recommended to use the IMAP as it provides the ability to manage emails using multiple devices or email clients, while allowing access to emails that are already read.
```
Comment (shafreenAnfar) [inline examples/receive-email-using-service/receive_email_using_service.md:5]:
```suggestion
An `email:Service` which is attached to an `email:ImapListener` can receive emails from an email server via IMAP protocol. An `email:ImapListener` can be initialized by providing the hostname, username, and password. Once connected, `onMessage` method will be invoked whenever an email is read from the email server. If there is an error while reading the data from the server, `onError` method is invoked with relevant error details. If the email server supports both POP3 and IMAP, it is recommended to use the IMAP as it provides the ability to manage emails using multiple devices or email clients, while allowing access to emails that are already read. 
```

File: examples/receive-email-using-client/receive_email_using_client.bal (hunk 3 above)
Comment (ayeshLK) [reply examples/receive-email-using-client/receive_email_using_client.bal]: Fixed with [5544886608e99affc49d1c303076d2eb039a0e08](https://github.com/ballerina-platform/ballerina-distribution/pull/3863/commits/5544886608e99affc49d1c303076d2eb039a0e08)

**Review by shafreenAnfar — APPROVED** [review]

File: examples/receive-email-using-service/receive_email_using_service.md [hunk 6]
```diff
@@ -1,6 +1,10 @@
 # Email service - Receive email
 
-The email service is used to receive (with POP3 or IMAP4) emails using the SSL or STARTTLS protocols. This sample includes receiving emails service attached to a POP3 listener with default configurations over SSL using the default ports. To use IMAP4 the user can use `IMAP listener`.
+The `email:Service` received messages from an email server via IMAP4 using the `email:ImapListener`.
+
+An `email:Service` which is attached to an `email:ImapListener` can receive emails from an email server via IMAP protocol. An `email:ImapListener` can be initialized by providing the hostname, username, and password. Once connected, `onMessage` method will be invoked whenever an email is read from the email server. If there is an error while reading the data from the server, `onError` method is invoked with relevant error details. If the email server supports both POP3 and IMAP, it is recommended to use the IMAP as it provides the ability to manage emails using multiple devices or email clients, while allowing access to emails that are already read.
```
Comment (shafreenAnfar) [inline examples/receive-email-using-service/receive_email_using_service.md:5]:
```suggestion
An `email:Service` which is attached to an `email:ImapListener` can receive emails from an email server via IMAP protocol. An `email:ImapListener` can be initialized by providing the hostname, username, and password. Once connected, `onMessage` method is invoked whenever an email is read from the email server. If there is an error while reading the data from the server, `onError` method is invoked with relevant error details. If the email server supports both POP3 and IMAP, it is recommended to use the IMAP as it provides the ability to manage emails using multiple devices or email clients, while allowing access to emails that are already read. 
```

File: examples/receive-email-using-service/receive_email_using_service.md [hunk 7]
```diff
@@ -1,6 +1,10 @@
 # Email service - Receive email
 
-The email service is used to receive (with POP3 or IMAP4) emails using the SSL or STARTTLS protocols. This sample includes receiving emails service attached to a POP3 listener with default configurations over SSL using the default ports. To use IMAP4 the user can use `IMAP listener`.
+The `email:Service` received messages from an email server via IMAP4 using the `email:ImapListener`.
+
```
Comment (shafreenAnfar) [inline examples/receive-email-using-service/receive_email_using_service.md:4]: Let's merge into one paragraph.

Comment (ayeshLK) [reply examples/receive-email-using-service/receive_email_using_service.md]: Fixed with [62617353559328a0bc02af20e81bb9fe5e79be60](https://github.com/ballerina-platform/ballerina-distribution/pull/3863/commits/62617353559328a0bc02af20e81bb9fe5e79be60)

**Review by chamil321 — APPROVED** [review]

**Pull request merged**